controllers/trainingjob/trainingjob_controller.go

```diff
@@ -170,6 +176,25 @@ func (r *Reconciler) reconcileTrainingJob(ctx reconcileRequestContext) error {
 		}
 	}
 
+	// Check  sagemaker describe output for each debug jobs and
+	// generates the update if k8s status for corresponding debug jobs differ.
+	// Initially status will be empty so we will generate update for all debug jobs
+	if len(ctx.TrainingJobDescription.DebugRuleEvaluationStatuses) != len(ctx.TrainingJob.Status.DebugRuleEvaluationStatuses) {
+		for _, _ = range ctx.TrainingJobDescription.DebugRuleEvaluationStatuses {
+			if err = r.addDebugRuleEvaluationStatusesToStatus(ctx); err != nil {
```

Comment:
Does this loop run `addDebugRuleEvaluationStatusesToStatus` for every status? Wouldn't this be adding them a total of `len(ctx.TrainingJobDescription.DebugRuleEvaluationStatuses)` times? From our definition of that method, it adds all of them at once, so we only need to run it once.

Reply:
I see.